### DISCUSSION TITLE
New Name/Icon

### DISCUSSION THREAD

**Discussion by polymorphicshade** [polymorphicshade/NewPipe#164] [Ideas]
As you all might know, I don't like the current name of this fork. Given my new intentions with development of this fork (see [here](https://www.reddit.com/r/NewPipe/comments/tg2spq/an_update_about_my_fork/) for reference) I want to rename this fork to something unique and recognizable.

The goal of the name is to help distinguish this fork from the original NewPipe.

So far the top features unique to this fork will be:
* [SponsorBlock](https://sponsor.ajay.app/) (including submissions and whatnot)
* [Return YouTube Dislike](https://www.returnyoutubedislike.com/) (including submitting dislikes)
* YouTube login functionality (liking, commenting, recommended content, etc)

For reference, here is a list of names suggested from the original issue I created back in 2020:
<details>
<summary>(click to expand)</summary>

* BlockPipe
* PipeFilter
* UnPipe 
* finerpipe
* Freshpipe
* NewerPipe
* SBPipe
* NewpipePlus
* SponsorPipe
* NewPipesOnTheBlock
* NewPipeS 
* SkipPipe 
* WarpPipe
* Tubular
* LibrePipe
* LiberaTube
* BetterPipe
* NeoPipe
* Newtube
* NPX (NewPipe eXtended)
* PimpedPipe
* AltPipe
* MeTube
* Fadtube
* Mytube
* Neohub
* Newatch
* Plumber
* Floodgate
* NewPipeX
* CleanPipe
* ClearPipe
* Chenille
* PolyPipe

</details>

Personally, I'm leaning towards the following:
* Tubular
* YouPipe/YourPipe (thought of this one as of creating this discussion topic)
* Floodgate

Though thinking about it... we probably want to avoid the word "Pipe" if possible.

As for the icon, I really like [this](https://user-images.githubusercontent.com/24195572/136754224-bfae6c52-225d-4906-b1f0-653786b61d82.png) one from @Mhowser, though I'm still totally open to ideas.

Discuss below. At some point I'll take the tally of "likes" to consideration when choosing a name.

**Comment by Mhowser** [comment]
Thanks Poly for liking the icon I made! I can adjust it further if needed.

Edit : I am also in favor of renaming to Tubular

**Comment by joelchrono12** [comment]
Floodgate for the win!

**Comment by DannyDarth** [comment]
Would this be the right place to ask a suggestion? If you are going to bring login back, Will there be a suggested videos that are recommended to you come back too? Like the Home on youtube?

**Reply by polymorphicshade** [reply]
This isn't the place to request features, but to answer your question, yes having suggested videos is part of the plan.

**Comment by yephny** [comment]
Login seems to be an extreme deviation from NewPipe. While sponsor block was on the gray area in how it matched the NewPipe project goals, it was rejected with creators, not users, in mind. Adding login, however, would be directly against NewPipe's privacy goals. I understand that this fork does not follow NewPipe's principles, but login still seems to be taking it too far. Does it imply that this fork will never be on the official F-Droid repository? Does it stem from the shutdown of Vanced, which should force many users to seek a 3rd party YouTube Android client that can log in?

**Reply by polymorphicshade** [reply]
Yes if you read my reddit post I linked above, I feel like the value of logging in exceeds the value of privacy. However this does not mean you won't be able to use NewPipe without logging in. I want to keep NewPipe how it is, except add the optional functionality of a YouTube login.

**Reply by infinitewaveparticle** [reply]
@polymorphicshade Not sure how much you follow feature requests on the upstream, but I would like for you to read this one I made here:

https://github.com/TeamNewPipe/NewPipe/issues/6762

Please take the time to read all of my comments and follow the links that I put there. If you're able to do what I suggest, then you'll satisfy most of the privacy concerns as well. Of course, just like the app I reference in the Feature Request, you could also offer the option to login using the user's personal gmail. I hope you take the time to see how the developer achieved what he did, as the app continues to work as intended to this day. It's pretty impressive, but at the same time it just makes perfect sense. Let me know if you're able to use his code from Gitlab to make a similar set of login options. Of course, you'd need to use a server like he does, but perhaps you could even reach out to him and see if he'd be willing to share the server he has set up (which would hopefully be cheaper than running one yourself). I'd be willing to pay (donate) for this option to help fund the server. If this doesn't make sense just read the link above and it will 🙂

**Reply by je-vv** [reply]
Oh Gosh, I hope not, :(  Even if being an opt in feature, I'd rather stay out of it.  Having any code authenticating with google sounds terrible to me, :(

But I guess not all of us think the same...

**Comment by Mhowser** [comment]
This is why I like this fork, it gives more power to the users.

**Comment by Qrbaker** [comment]
Re: your reddit post about becoming a vanced alternative, with YouTube logins, etc.

That would open a massive can o' worms for Google Play services. I'm sure you're considering that, but I want everyone else to be aware that this would mean potentially handing over all the data that NewPipe was designed to not do. You can alleviate this somewhat with MicroG, but that then requires users installing a second APK purely to shim their play API interactions.

And finally, it looks like a more direct successor to vanced is in rapid development, this time as a [patcher for the YT app directly](https://github.com/ReVancedTeam). The release and success is not guaranteed, of course, but it reflects an important distinction.

I suspect that (until recently with the Vanced exodus) the userbases of these two apps are distinct. Coding aside, I would encourage @polymorphicshade to carefully consider what catering to a Vanced-like audience entails. Expect a lot more casual users -- not inherently bad, of course, but this will mean a lot of GitHub issue spam and takes a non-trivial effort to wade through. It also will draw an audience with a very different attitude. I do generally think that this project fork is mostly used by individuals that ultimately agree with upstream principles regarding privacy, if not the choices and results derived therefrom.

The community this may foster will almost certainly be more demanding, less contributing, and will have less of an understanding of the open source flow and process. This means a lot more friction, and I'd hate to see burnout.

---

On the branding/naming side of things, I wholehearted support these changes. I think distancing from upstream will save both them and this group headaches. Tubular has a fun ring to it, and the logo in question looks nice!

**Reply by polymorphicshade** [reply]
All good points. As for using Google Play services or MicroG, I've recently found a way to login to YouTube and use the cookies for web requests, like getting Home page recommendations. None of this relies on Google Play services nor MicroG as I understand it - just a simple web page login and cookie retention.

Also I do expect a lot more casual users, and thus more "casual" bug-reports haha... Not really something I care about that this moment in time.

**Reply by Qrbaker** [reply]
> I've recently found a way to login to YouTube and use the cookies for web requests

If it can be done reliably and doesn't result in a cat-and-mouse game where the app breaks randomly until an update gets pushed, ok!

**Reply by je-vv** [reply]
> All good points. As for using Google Play services or MicroG, I've recently found a way to login to YouTube and use the cookies for web requests, like getting Home page recommendations. None of this relies on Google Play services nor MicroG as I understand it - just a simple web page login and cookie retention.

So this sounds like being a path, something getting investigated, and sooner or later it's something you're willing to enable.  That's so sad to hear from my side.  Thanks for showing newpipe with sponsorblock was possible, and such a fork was possible.

**Comment by polymorphicshade** [comment]
So far it seems Tubular is the most popular name. I'm going to give this discussion a week or so until I decide on a name, but until then I'll be using Tubular and the new branding, unless folks think otherwise.

I'll be creating a Discord channel with the branding for a more casual place to discuss features and what's going on with development.

**Reply by ghost** [reply]
I personally like FloodGate, has a professional feel to it, Tubular is too joke-y for me.

**Reply by joelchrono12** [reply]
I wholeheartedly agree with you. Yes I know this is a personal/community project and its nothing too corporate or whatever, but still.

**Reply by ghost** [reply]
I should have said concise and simple rather than professional, NewPipe is named in a similar way instead of some joke name. If the idea is to have a name like GIMP where it's both somewhat of a joke name and a concise one when it's not an acronym then Tubular doesn't do that either.

**Reply by infinitewaveparticle** [reply]
Ones I've come up with just now (in order of my brain spewing them out):

TubeBlock
Tublock
Tublock Shacore (hehe)
WeTube
PipeWerx
Alltube
TubeTime
TubeStone
Nospo
CleanTube
BesTube
Tubule
NoadTube (as in No Ad Tube)
NoadsTube (as in No Ads Tube)
TubeNoad
Tubesby (as in Tube SBY, or Tube SponsorBlock YouTube)
Tubesbay (as in Tube SB AY, or Tube SponsorBlock Ad-free YouTube)
Tubesbi (as in Tube SBI, or Tube SponsorBlock Implementation)
Pipesbi/Pipesby/Pipesbay
Tubopoly
Polytub
Toobsby
Toobular
basically anything above with Toob
Tubex
Tubensis
Tuber
Yerterb

No idea if any of these were used by other projects or suggested before... Just throwing stuff out there

**Reply by ghost** [reply]
I rather like CleanTube actually, to me it does represent the project well. It's also concise and simple.

**Comment by Mhowser** [comment]
I still stand by Tubular because you know, we are _surfing the web!_

**Reply by joelchrono12** [reply]
Tubular ain't a web browser tho :P

**Reply by Mhowser** [reply]
Sure but the videos are hosted on the Internet regardless

**Comment by KIKOKI** [comment]
is this the right place to suggest something? Regarding the popup player, I think you should change the function of the "fulls creen"button. As of now whenever the button is pressed, It goes on landscaped full screen player. Maybe mimic how Vanced does this by just going back to the main app view/player(portrait) instead of going full screen landscaped.

**Reply by infinitewaveparticle** [reply]
You should create a new issue, and if it's something that also happens in the standard NewPipe app, then you'll need to open the issue in their repo, not here.

**Comment by WreckingBANG** [comment]
I like PolyPipe

**Reply by infinitewaveparticle** [reply]
Tubopoly

**Comment by s2r** [comment]
Heyo, well this is basically a enhanced newpipe, nothing wrong with that, the "extra youtube functionality" would be great, but this app is like 95% newpipe, so the name should be more related, i also think youtube is more aggressive with tube like names... something like "neopipe" or "ultimapipe" (in the sense that newpipe is "proto"pipe) or a version of pipe, like "newline" or "newmain" i think those are better... than tubular, tubular sounds like california surfer slang, but thats me...

Regarding the icon... hummm yeah you can go with any type of play icon or maybe a version of new pipe like this one i made, just a version of newpipe and random background color (please note my photoshop skills are rudimentary) ![newnewpipe](https://user-images.githubusercontent.com/2265583/163670477-6815be25-a533-4d0c-abe6-5efe2c638cde.png) , have a good one and thanks for the awesome work!

**Comment by polymorphicshade** [comment]
So I've decided to go with the name Tubular, and I'll be locking this discussion. I've created a [Discord server](https://discord.gg/Jgb7kuVfrt) for those who are interested.
